pkg/config/store/convert.go

```diff
@@ -24,6 +24,28 @@ import (
 	"github.com/golang/glog"
 )
 
+const (
+	ruleKind      = "rule"
+	selectorField = "selector"
+	matchField    = "match"
+)
+
+// warnDeprecationAndFix warns users about deprecated fields.
+// It maps the field into new name.
+func warnDeprecationAndFix(key Key, spec map[string]interface{}) map[string]interface{} {
+	if key.Kind != ruleKind {
+		return spec
+	}
+	sel := spec[selectorField]
+	if sel == nil {
+		return spec
+	}
+	glog.Warningf("Deprecated field 'selector' used in %s. Use 'match' instead.", key)
+	spec[matchField] = sel
```

Comment:
should we check if `matchField` already exists in the spec?  It's very unlikely to happen, but it overwrites the match field if both exist.